src/main/java/edu/hm/hafner/analysis/ast/Ast.java

```diff
@@ -0,0 +1,520 @@
+        for (byte b : digest) {
+            result.append(String.format("%02x", b));
+        }
+        return result.toString();
+    }
+
+    private MessageDigest createMessageDigest() {
+        try {
+            return MessageDigest.getInstance(HASH_ALGORITHM);
+        }
+        catch (NoSuchAlgorithmException e) {
+            throw new IllegalArgumentException("Requested algorithm not found: " + HASH_ALGORITHM);
+        }
+    }
+
+    private DetailAST objBlock;
+
+    /**
+     * Returns the objBlock of the abstract syntax tree.
+     *
+     * @param topRoot
+     *         the highest root of the ast
+     *
+     * @return the objblock
+     */
+    protected DetailAST getObjBlock(final DetailAST topRoot) {
+        calcObjBlock(topRoot, 0);
+
+        return objBlock;
+    }
+
+    /**
+     * Calculates the OBJBLOCK of the ast.
+     *
+     * @param topRoot
+     *         the highest root of the ast
+     * @param start
+     *         which OBJBLOCK should be found? For example if you want have the first OBJBLOCK, you have to set the
+     *         counter to 0.
+     */
+    protected void calcObjBlock(final DetailAST topRoot, final int start) {
+        if (topRoot != null) {
+            int counter = start;
+            if (topRoot.getType() == TokenTypes.OBJBLOCK && counter == 0) {
+                objBlock = topRoot;
+                counter++;
+            }
+            if (topRoot.getFirstChild() != null) {
+                calcObjBlock(topRoot.getFirstChild(), counter);
+            }
+            if (topRoot.getNextSibling() != null) {
+                calcObjBlock(topRoot.getNextSibling(), counter);
+            }
+        }
+    }
+
+    /**
+     * Calculates the last sibling of the given ast-element.
+     *
+     * @param element
+     *         the current element in the ast
+     *
+     * @return the last element.
+     */
+    protected DetailAST getLastSibling(final DetailAST element) {
+        if (element.getNextSibling() != null) {
+            return getLastSibling(element.getNextSibling());
+        }
+        return element;
+    }
+
+    /**
+     * Calculates the last line number of the abstract syntax tree.
+     *
+     * @return Returns the last linenumber of the ast.
+     */
+    public int getLastLineNumber() {
+        return getLastSibling(getObjBlock(root).getFirstChild()).getLineNo();
+    }
+
+    // Map<nameOfConstant, value>
+    private final Map<DetailAST, DetailAST> constants = new HashMap<>();
+
+    /**
+     * Returns {@code true}, if the element is a constant, otherwise returns {@code false}.
+     *
+     * @param element
+     *         the element
+     *
+     * @return {@code true}, if the element is a constant, otherwise returns {@code false}.
+     */
+    protected boolean isConstant(final DetailAST element) {
+        if (element.getType() != TokenTypes.VARIABLE_DEF) {
+            return false;
+        }
+        else if (element.getChildCount() != 5) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid using Literals in Conditional Statements](https://app.codacy.com/manual/uhafner/analysis-model/pullRequest?prid=5019963)